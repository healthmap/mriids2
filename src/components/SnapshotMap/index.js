import React, { useState } from "react";
import { connect } from "react-redux";
import {
  ComposableMap,
  Geographies,
  Geography,
  ZoomableGroup,
} from "react-simple-maps";

import { SnapshotMapContainer } from "../styled-components/MapContainers";
import ViewToggle from "../ViewToggle";
import SnapshotMapCaseCountLegend from "../SnapshotMapCaseCountLegend";
import MapZoomButtons from "../MapZoomButtons";
import {
  getScale,
  getSnapshotColor,
  getSnapshotProjectionsColor,
} from "../../utils/snapshotMapHelpers";
import { getEbolaCountriesCaseCounts } from "../../utils/ebolaDataHelpers";

const SnapshotMap = ({ ebolaData, filters }) => {
  const [zoomLevel, setZoomLevel] = useState(5);
  const [showCaseCounts, changeShowCaseCounts] = useState(true);

  const changeZoomLevel = (newZoomValue = 0) => {
    setZoomLevel(newZoomValue);
  };

  const getGeographyFillColor = (geoProperties) => {
    const ebolaCountries = ["Guinea", "Liberia", "Sierra Leone"];
    // If the NAME of the geography is in the ebolaCountries array, execute this block.
    if (ebolaCountries.includes(geoProperties.NAME)) {
      // Get the case count for the 3 ebolaCountries.
      const ebolaCountriesCaseCounts = getEbolaCountriesCaseCounts(
        ebolaData,
        filters
      );
      const scale = getScale(ebolaCountriesCaseCounts);
      const percentage = ebolaCountriesCaseCounts[geoProperties.NAME] / scale;
      // If projections are enabled, return the color value using the getSnapshotProjectionsColor function.
      // Otherwise return the color value using the getSnapshotColor function.
      return filters.projection
        ? getSnapshotProjectionsColor(percentage)
        : getSnapshotColor(percentage);
    } else {
      // If the NAME of the geography is not in the ebolaCountries array, add the fill color below.
      return "#FCF1DD";
    }
  };

  const showHideCaseCounts = () => {
    changeShowCaseCounts(!showCaseCounts);
  };

  return (
    <SnapshotMapContainer>
      <ViewToggle />
      <ComposableMap
        style={{ backgroundColor: "#D6E4EE" }}
        stroke="#000000"
        strokeWidth={0.05}
      >
        <ZoomableGroup zoom={zoomLevel}>
          <Geographies geography="mapData/world-50m-simplified.json">
            {({ geographies }) =>
              geographies.map((geo) => (
                <Geography
                  key={geo.rsmKey}
                  geography={geo}
                  style={{
                    default: {
                      fill: getGeographyFillColor(geo.properties),
                      opacity: 1,
                      transition: "all .5s ease",
                    },
                    hover: {
                      fill: getGeographyFillColor(geo.properties),
                      opacity: 0.8,
                      transition: "all .5s ease",
                    },
                    pressed: {
                      fill: getGeographyFillColor(geo.properties),
                      opacity: 0.8,
                      transition: "all .5s ease",
                    },
                  }}
                />
              ))
            }
          </Geographies>
        </ZoomableGroup>
      </ComposableMap>
      <SnapshotMapCaseCountLegend
        showCaseCounts={showCaseCounts}
        toggleCaseCountFunction={showHideCaseCounts}
      />
      <MapZoomButtons
        viewPortZoom={zoomLevel}
        changeZoomFunction={changeZoomLevel}
      />
    </SnapshotMapContainer>
  );
};

const mapStateToProps = (state) => ({
  ebolaData: state.ebola.ebolaData.data,
  filters: state.filters,
});

export default connect(mapStateToProps)(SnapshotMap);
