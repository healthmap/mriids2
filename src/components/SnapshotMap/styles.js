import styled from "styled-components";
import ReactTooltip from "react-tooltip";

export const SnapshotMapContainer = styled.div`
  flex: none;
  position: relative;
  top: ${({ isProjectionsBannerDisplayed, theme: { bannerHeight } }) =>
    isProjectionsBannerDisplayed ? bannerHeight : 0};
  padding-left: ${(props) => props.theme.sidebarWidth};
  height: calc(100vh / 2);
  min-height: 40rem;
  overflow: hidden;
`;

export const StyledTooltip = styled(ReactTooltip)`
  && {
    color: ${(props) => props.theme.textColor};
    background-color: ${(props) => props.theme.background1};
    opacity: 0.95 !important;
    border-radius: 3px;
  }
  &:after {
    border-top-color: ${(props) => props.theme.background1} !important;
  }
`;
