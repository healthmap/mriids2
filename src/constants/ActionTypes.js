export const FETCH_EBOLA_DATA_REQUEST = "FETCH_EBOLA_DATA_REQUEST";
export const FETCH_EBOLA_DATA_SUCCESS = "FETCH_EBOLA_DATA_SUCCESS";
export const FETCH_EBOLA_DATA_FAILURE = "FETCH_EBOLA_DATA_FAILURE";

export const FETCH_EBOLA_DATA_COMBINED_REQUEST =
  "FETCH_EBOLA_DATA_COMBINED_REQUEST";
export const FETCH_EBOLA_DATA_COMBINED_SUCCESS =
  "FETCH_EBOLA_DATA_COMBINED_SUCCESS";
export const FETCH_EBOLA_DATA_COMBINED_FAILURE =
  "FETCH_EBOLA_DATA_COMBINED_FAILURE";

export const FETCH_RISK_DATA_REQUEST = "FETCH_RISK_DATA_REQUEST";
export const FETCH_RISK_DATA_SUCCESS = "FETCH_RISK_DATA_SUCCESS";
export const FETCH_RISK_DATA_FAILURE = "FETCH_RISK_DATA_FAILURE";

export const FETCH_COVID_CASE_COUNT_DATA_REQUEST =
  "FETCH_COVID_CASE_COUNT_DATA_REQUEST";
export const FETCH_COVID_CASE_COUNT_DATA_SUCCESS =
  "FETCH_COVID_CASE_COUNT_DATA_SUCCESS";
export const FETCH_COVID_CASE_COUNT_DATA_FAILURE =
  "FETCH_COVID_CASE_COUNT_DATA_FAILURE";

export const FETCH_COVID_DATA_COMBINED_REQUEST =
  "FETCH_COVID_DATA_COMBINED_REQUEST";
export const FETCH_COVID_DATA_COMBINED_SUCCESS =
  "FETCH_COVID_DATA_COMBINED_SUCCESS";
export const FETCH_COVID_DATA_COMBINED_FAILURE =
  "FETCH_COVID_DATA_COMBINED_FAILURE";

// These are used for updating the filters
export const CHANGE_COUNTRY = "CHANGE_COUNTRY";
export const CHANGE_OUTBREAK = "CHANGE_OUTBREAK";
export const CHANGE_DATE_RANGE = "CHANGE_DATE_RANGE";
export const CHANGE_VIEW = "CHANGE_VIEW";
export const CHANGE_PROJECTION = "CHANGE_PROJECTION";
