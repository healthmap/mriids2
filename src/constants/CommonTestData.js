export const reduxInitialState = {
  ebola: {
    ebolaData: {
      isFetching: 0,
      data: {},
      error: {},
    },
    ebolaDataCombined: {
      isFetching: 0,
      data: {},
      error: {},
    },
    riskData: {
      isFetching: 0,
      data: {},
      error: {},
    },
  },
  covid: {
    caseCounts: {
      isFetching: 0,
      data: [],
      error: {},
    },
    deathCounts: {
      isFetching: 0,
      data: [],
      error: {},
    },
  },
  filters: {
    country: "All",
    outbreak: "Ebola Outbreak",
    view: "snapshot",
    projection: false,
    dateRange: { from: new Date(2014, 9, 1), to: new Date(2016, 1, 20) },
  },
};

// Same as the reduxInitialState, but with the 'risk' view in the filters.
export const riskViewState = {
  ...reduxInitialState,
  filters: {
    ...reduxInitialState.filters,
    view: "risk",
  },
};
