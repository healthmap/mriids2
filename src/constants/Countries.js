export const ebolaOutbreakCountries = ["Guinea", "Liberia", "Sierra Leone"];

export const allCountries = [
  "Afghanistan",
  "Albania",
  "Algeria",
  "Andorra",
  "Angola",
  "Antigua and Barbuda",
  "Argentina",
  "Armenia",
  "Aruba",
  "Australia",
  "Austria",
  "Azerbaijan",
  "Bahamas",
  "Bahrain",
  "Bangladesh",
  "Barbados",
  "Belarus",
  "Belgium",
  "Belize",
  "Benin",
  "Bhutan",
  "Bolivia",
  "Bosnia and Herzegovina",
  "Botswana",
  "Brazil",
  "British Virgin Islands",
  "Brunei Darussalam",
  "Bulgaria",
  "Burkina Faso",
  "Burundi",
  "Cambodia",
  "Cameroon",
  "Canada",
  "Cape Verde",
  "Cayman Islands",
  "Central African Republic",
  "Chad",
  "Chile",
  "China",
  "Colombia",
  "Comoros",
  "Congo",
  "Costa Rica",
  "Cote dIvoire",
  "Croatia",
  "Cuba",
  "Cyprus",
  "Czechia",
  "Democratic Republic of the Congo",
  "Denmark",
  "Djibouti",
  "Dominica",
  "Dominican Republic",
  "Ecuador",
  "Egypt",
  "El Salvador",
  "Equatorial Guinea",
  "Eritrea",
  "Estonia",
  "Eswatini",
  "Ethiopia",
  "Fiji",
  "Finland",
  "France",
  "Gabon",
  "Gambia",
  "Georgia",
  "Germany",
  "Ghana",
  "Greece",
  "Greenland",
  "Grenada",
  "Guatemala",
  "Guinea",
  "Guinea Bissau",
  "Guyana",
  "Haiti",
  "Holy See",
  "Honduras",
  "Hungary",
  "Iceland",
  "India",
  "Indonesia",
  "Iran",
  "Iraq",
  "Ireland",
  "Israel",
  "Italy",
  "Jamaica",
  "Japan",
  "Jordan",
  "Kazakhstan",
  "Kenya",
  "Kosovo",
  "Kuwait",
  "Kyrgyzstan",
  "Laos",
  "Latvia",
  "Lebanon",
  "Lesotho",
  "Liberia",
  "Libya",
  "Liechtenstein",
  "Lithuania",
  "Luxembourg",
  "Madagascar",
  "Malawi",
  "Malaysia",
  "Maldives",
  "Mali",
  "Malta",
  "Marshall Islands",
  "Mauritania",
  "Mauritius",
  "Mexico",
  "Moldova",
  "Monaco",
  "Mongolia",
  "Montenegro",
  "Morocco",
  "Mozambique",
  "Myanmar",
  "Namibia",
  "Nepal",
  "Netherlands",
  "New Zealand",
  "Nicaragua",
  "Niger",
  "Nigeria",
  "North Macedonia",
  "Norway",
  "Oman",
  "Pakistan",
  "Panama",
  "Papua New Guinea",
  "Paraguay",
  "Peru",
  "Philippines",
  "Poland",
  "Portugal",
  "Puerto Rico",
  "Qatar",
  "Romania",
  "Russia",
  "Rwanda",
  "Saint Kitts and Nevis",
  "Saint Lucia",
  "Saint Vincent and the Grenadines",
  "Samoa",
  "San Marino",
  "Sao Tome and Principe",
  "Saudi Arabia",
  "Senegal",
  "Serbia",
  "Seychelles",
  "Sierra Leone",
  "Singapore",
  "Slovakia",
  "Slovenia",
  "Solomon Islands",
  "Somalia",
  "South Africa",
  "South Korea",
  "South Sudan",
  "Spain",
  "Sri Lanka",
  "Sudan",
  "Suriname",
  "Sweden",
  "Switzerland",
  "Syria",
  "Taiwan",
  "Tajikistan",
  "Thailand",
  "Timor Leste",
  "Togo",
  "Trinidad and Tobago",
  "Tunisia",
  "Turkey",
  "Uganda",
  "Ukraine",
  "United Arab Emirates",
  "United Kingdom",
  "United Republic of Tanzania",
  "United States of America",
  "Uruguay",
  "Uzbekistan",
  "Vanuatu",
  "Venezuela",
  "Vietnam",
  "Yemen",
  "Zambia",
  "Zimbabwe",
];
