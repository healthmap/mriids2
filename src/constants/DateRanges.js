export const ebolaInitialDateRange = {
  from: new Date(2014, 9, 1),
  to: new Date(2016, 1, 20),
};
