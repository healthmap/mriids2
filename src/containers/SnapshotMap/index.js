import React, { Component } from "react";
import ReactMapGL from "react-map-gl";

import { SnapshotMapContainer } from "../../components/styled-components/MapContainers";
import ViewToggle from "../../components/ViewToggle";
import SnapshotMapCaseCountLegend from "../../components/SnapshotMapCaseCountLegend";
import MapZoomButtons from "../../components/MapZoomButtons";

class SnapshotMap extends Component {
  constructor(props) {
    super(props);
    this.state = {
      viewport: {
        width: "100%",
        height: "100%",
        latitude: 8.555216,
        longitude: -11.322184,
        zoom: 5,
        minZoom: 2,
        pitch: 0,
        bearing: 0,
      },
      settings: {
        dragPan: true,
        dragRotate: true,
        scrollZoom: true,
        touchZoom: true,
        touchRotate: true,
        keyboard: true,
        doubleClickZoom: true,
      },
      showCaseCounts: true,
    };
  }

  onViewportChange = (viewport) => {
    this.setState({ viewport });
  };

  changeZoomLevel = (newZoomValue = 0) => {
    this.setState((currentState) => ({
      viewport: {
        ...currentState.viewport,
        zoom: newZoomValue,
      },
    }));
  };

  showHideCaseCounts = () => {
    this.setState((currentState) => ({
      showCaseCounts: !currentState.showCaseCounts,
    }));
  };

  render() {
    return (
      <SnapshotMapContainer>
        <ViewToggle />
        <ReactMapGL
          {...this.state.viewport}
          onViewportChange={(viewport) => this.onViewportChange(viewport)}
          mapStyle="mapbox://styles/compepi/cjnxgpr991b6h2rpcvqmh5j4f"
        />
        <SnapshotMapCaseCountLegend
          showCaseCounts={this.state.showCaseCounts}
          toggleCaseCountFunction={this.showHideCaseCounts}
        />
        <MapZoomButtons
          viewPortZoom={this.state.viewport.zoom}
          changeZoomFunction={this.changeZoomLevel}
        />
      </SnapshotMapContainer>
    );
  }
}

export default SnapshotMap;
