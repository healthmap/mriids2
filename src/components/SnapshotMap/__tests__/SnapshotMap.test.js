import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { reduxInitialState } from "../../../constants/CommonTestData";
import SnapshotMap from "../index";

const mockStore = configureStore([thunk]);

describe("Tests for the connected SnapshotMap component with reduxInitialState", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(reduxInitialState);

    component = renderer.create(
      <Provider store={store}>
        <SnapshotMap />
      </Provider>
    );
  });
  test("SnapshotMap component should have 5 child components", () => {
    expect(component.toJSON().children).toHaveLength(5);
  });
});
