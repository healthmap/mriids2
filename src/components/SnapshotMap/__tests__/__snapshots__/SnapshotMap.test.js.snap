// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the connected SnapshotMap component with reduxInitialState should render with given state from Redux store 1`] = `
.c0 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  position: relative;
  height: 50rem;
  overflow: hidden;
}

.c2 {
  min-width: 5rem;
  padding: 0.4rem 0.8rem;
  font-weight: bold;
  text-transform: uppercase;
  border: 1px solid #000;
  border-top: 0;
  cursor: pointer;
  background-color: #fff;
  -webkit-transition: background-color 0.3s ease;
  transition: background-color 0.3s ease;
}

.c2:first-child {
  border-bottom-left-radius: 0.3rem;
}

.c2:last-child {
  border-bottom-right-radius: 0.3rem;
}

.c2:hover,
.c2:focus {
  background-color: #ececec;
}

.c2.is-active {
  color: #fff;
  background-color: #000;
}

.c1 {
  position: absolute;
  left: 50%;
  z-index: 2;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 2px;
}

.c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 2em;
  height: 2em;
  margin-right: 0.5em;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #666;
}

.c9 {
  position: absolute;
  bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 1em;
  padding: 1em 0.5em 0 0.5em;
  border-top: 1px solid #ccc;
  cursor: pointer;
}

.c9 label {
  margin-left: 0.8em;
  line-height: 1.1;
  color: #666;
  cursor: pointer;
}

.c3 {
  position: absolute;
  top: 2rem;
  left: 2rem;
  bottom: 7rem;
  width: 9rem;
  overflow: hidden;
}

.c3 > div {
  width: 100%;
  height: 100%;
}

.c5 {
  height: calc(100% - 7rem);
  overflow: auto;
}

.c4 {
  display: inline-block;
  padding: 1rem;
  box-shadow: 1px 1px 6px 0px rgba(0,0,0,0.1);
  border-radius: 0.5rem;
  background-color: #fff;
}

.c10 {
  position: absolute;
  right: 10px;
  bottom: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  box-shadow: 1px 1px 6px 0px rgba(0,0,0,0.1);
  border-radius: 0.5rem;
  background-color: #fff;
  overflow: hidden;
}

.c11 {
  display: block;
  padding: 0.8em;
  color: #000;
  font-size: 16px;
  line-height: 0.8;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  background-color: #fff;
  border: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
}

.c11:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
}

.c11:active {
  background-image: none;
  outline: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c11:last-child {
  border-top: 1px solid #ececec;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      className="c2 is-active"
      onClick={[Function]}
    >
      Snapshot
    </button>
    <button
      className="c2"
      onClick={[Function]}
    >
      Risk
    </button>
  </div>
  <svg
    className="rsm-svg "
    stroke="#000000"
    strokeWidth={0.05}
    style={
      Object {
        "backgroundColor": "#D6E4EE",
      }
    }
    viewBox="0 0 800 600"
  >
    <g>
      <rect
        fill="transparent"
        height={600}
        width={800}
      />
      <g
        className="rsm-zoomable-group undefined"
        transform="translate(0 0) scale(1)"
      >
        <g
          className="rsm-geographies "
        />
      </g>
    </g>
  </svg>
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      <h3>
        Case counts
      </h3>
      <div
        className="c5"
      >
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#E23D4A",
              }
            }
          />
          <div
            className="c8"
          >
            20
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#E23D4A",
              }
            }
          />
          <div
            className="c8"
          >
            18
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#E55259",
              }
            }
          />
          <div
            className="c8"
          >
            16
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#E86769",
              }
            }
          />
          <div
            className="c8"
          >
            13
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#EB7C77",
              }
            }
          />
          <div
            className="c8"
          >
            11
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#EE9187",
              }
            }
          />
          <div
            className="c8"
          >
            9
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#F1A697",
              }
            }
          />
          <div
            className="c8"
          >
            7
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#F5BCA7",
              }
            }
          />
          <div
            className="c8"
          >
            4
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#F8D1B6",
              }
            }
          />
          <div
            className="c8"
          >
            2
          </div>
        </div>
        <div
          className="c6"
        >
          <div
            className="c7"
            style={
              Object {
                "backgroundColor": "#FDF1DD",
              }
            }
          />
          <div
            className="c8"
          >
            0
          </div>
        </div>
      </div>
      <div
        className="c9"
        onClick={[Function]}
      >
        <svg
          alt="Hide Count"
          aria-hidden="true"
          height="20px"
          role="img"
          style={
            Object {
              "height": "20px",
              "width": "20px",
            }
          }
          viewBox="0 0 100 100"
          width="20px"
        >
          <g>
            <title>
              Hide Count
            </title>
            <path
              d="M75.3,30.1l11.6-11.6c1.4-1.4,1.4-3.9,0-5.4s-3.9-1.4-5.4,0L67.9,26.5c-5.9-2.6-12-4.2-18.1-4.2 c-8.5,0-17.1,3-25.1,7.6C15.6,35.3,7.2,42.6,0.4,49.9c6,6.7,12.9,13.3,20.8,18.5c1,0.6,2,1.3,3,1.9L13.7,80.7L13,81.4 c-1.4,1.4-1.4,3.9,0,5.4c1.4,1.6,3.9,1.4,5.4,0l0.7-0.7l0,0l12.3-12.3c5.7,2.3,11.9,3.7,18.5,3.7c10.8,0,20.4-3.7,28.7-9.2 c7.9-5.2,14.8-11.8,20.8-18.5C92.9,42.7,84.4,35.4,75.3,30.1z M32.7,62.1c-2.3-3.4-3.7-7.6-3.7-12c0-5.7,2.4-11,6.2-14.9 c4-4,9.3-6.2,14.9-6.2c4.4,0,8.6,1.4,12,3.7l-4.2,4.2c-2.3-1.4-5-2.2-7.9-2.2c-4.3,0-8.2,1.7-10.9,4.4c-2.7,2.7-4.4,6.6-4.4,10.9 c0,2.9,0.9,5.6,2.2,7.9L32.7,62.1z M64.9,64.9c-3.9,3.9-9.2,6.2-14.9,6.2c-4.4,0-8.6-1.4-12-3.7l4.2-4.2c2.3,1.4,5,2.2,7.9,2.2 c4.3,0,8.2-1.7,10.9-4.4c2.7-2.7,4.4-6.6,4.4-10.9c0-2.9-0.9-5.6-2.2-7.9l4.2-4.2c2.3,3.4,3.7,7.6,3.7,12 C71.1,55.8,68.8,61.1,64.9,64.9z"
              fill="#000000"
            />
          </g>
        </svg>
        <label>
          Hide Count
        </label>
      </div>
    </div>
  </div>
  <div
    className="c10"
  >
    <button
      className="c11"
      disabled={false}
      onClick={[Function]}
    >
      +
    </button>
    <button
      className="c11"
      disabled={false}
      onClick={[Function]}
    >
      -
    </button>
  </div>
</div>
`;
